### ISSUE TITLE
Structures not spawning on server at all...

### ISSUE THREAD

**Issue by Kilroy5150** [Ivorforce/RecurrentComplex#199]
Running Minecraft 1.10.2/Forge 18.3.2281 on an i7 machine (16GB memory).

For some reason i cannot fathom, it just won't spawn anything. I've been running this server for some time, just got the newest version for 1.10.2 (the one before that didn't work either). 

It will not spawn any structures in the village, nothing at all....I go between 2 and 3 villages and nothing....

I am using the AMP server launcher and have tried both General and Customized worlds to no avail...

Anyone?

**Comment by Ivorforce** [comment]
Try using /#list to see if any structures are registered first.

**Comment by Kilroy5150** [comment]
It gave me this:

16:22:57No Structures

**Comment by Kilroy5150** [comment]
Anyone? Anything? I really need to get this mod working. I've got a team that's chomping at the bit to play with it on the game...

**Comment by Ivorforce** [comment]
Yes, it's this again: 
java.nio.file.AccessDeniedException

See #154. Let me know if you have any issues setting the access rights correctly.

**Comment by Kilroy5150** [comment]
.I run the AMP launcher and it handles the whole process...no other mods are having this issue. There are around 66 mods running on my server. What would be causing yours not to have access rights?

What other folders does your mod try to write to so i can track down the issue?

I need to know what files it writes in order to track it down but it's odd that, out of 66 mods, yours is the only one with this issue...:(

**Comment by Ivorforce** [comment]
As described in #154, ReC has special scripts that read structures automatically from all mods (and itself). As far as I'm aware, no other mod does this - they all load their stuff referenced by jar resource, and thus don't need access rights to anything.

It's not the mod though that needs the rights, it's the process (Java) that MC runs on. It needs read access to everything in /mods and /structures.

**Comment by Kilroy5150** [comment]
Is it something as simple as the "give admin" checkbox in the comparability tab on java.exe?  I'm afraid i have less experience in running this sort of issue down as i have very little experience with the inner-workings of Java...

**Comment by Ivorforce** [comment]
That might do it if you do it for the minecraft application (not Java.exe, I believe) - I'm not sure how Windows handles privileges though. It might also be enough to check if the user you run MC with has read access to the mods folder (right click -> properties or something). @ArmandVestibule perhaps you want to weigh in with your fix?

**Comment by ArmandVestibule** [comment]
Sure. If AMP is anything like McMyAdmin, it should run as a service. That service will have an account linked to it and it should require a certain level of permissions in order to run Recurrent Complex successfully.

In my particular case, under Windows, my McMyAdmin service was running as Network Service instead of the Local System -- once I changed it to the latter, everything started to work.

There may be a more elegant way to provide the level of access the mod needs to function in this context, but that worked for me.

**Comment by Kilroy5150** [comment]
Well AMP is the successor to MyMcAdmin.....and making it into a local service crashed it permanently...i can't even put it back to a network service and make it run....i'm going to have to reinstall the whole instance again. AMP launches games a individual "instances" (their term, not mine)

**Comment by Ivorforce** [comment]
If it starts to crash and doesn't stop I'd definitely report this to the AMP team.

**Comment by Kilroy5150** [comment]
Yeah, that's like talking to a wall. When it comes to relating technical to lamen....they suck. They don't speak "simple" and they're somewhat conceited when you ask them to spell out what they need done. They sorta assume everyone talking to them should have their level of experience. While i'm not a novice with computers by a long shot....i don't code so they will tell me something that's a little above my level and basically i have to spend a while Googling to find out what they're talking about. 

AMP is their attempt to be "the" server launcher for all home server systems and, while it's decent, they put in wrong version of Minecraft on the list to select, have unnecessary options that don't even need to be there...

Man, i miss MyMcAdmin :)

**Comment by Kilroy5150** [comment]
Well removing it as a service certainly cleared up the issue...I've just got to find a way to get it running as a service (and local as you suggested) without it crashing the run. It's like a double-edged sword...AMP makes it easier but seems to cause constant issues with Minecraft. This thing is still in BETA but reporting any kind of issue to them gets them all twisted and they want to make it seem like you're doing something wrong so i will attempt to fix it on my end before reporting it to them...to kinda have a whole story to go on..

**Comment by ArmandVestibule** [comment]
Do they no longer distribute McMyAdmin 2? I still run it, myself -- this new approach of theirs seems too... kitchen sink, imo.

**Comment by Kilroy5150** [comment]
Dunno, i got talked into buying AMP for $5 because it was in Beta. It's got a lot that needs working out before i would consider it stable. Like i said, the put wrong version numbers on the Minecraft they have listed, they're not listed in any kind of credible order. The interface is very "vanilla" but they've added a lot of interesting feature. 

It's strange, after i reinstalled a new instance the problem went away. My best guess is that possibly AMP corrupted something. But again, it could run this time and not the next. This software is very glitchy and tends to change it's mind of what it wants to do from time to time. 

It does allow you to pile on the memory with the setup and someone gave me an I7 desktop with 16GB of RAM so i've got memory to spend i've just not figured out if it's worth it yet. I always hear Java is a big memory hog but, at the same time, Minecraft has so many obvious problems that could have been fixed (by the company) so long ago that it just kills me.

The community made Minecraft great when they started modding. I got so bored, so early, with Minecraft that i put it down for a couple of years and came back to see a flourishing modding community which had mods that gave it so much more of a personality and a lot of what you see on vanilla minecraft, now, is stuff that were mods before (like the random fishing affects, i remember that mod clearly a while back)
